lib/Cake/View/Helper/HtmlHelper.php

```diff
@@ -829,6 +829,15 @@ public function image($path, $options = array()) {
 			unset($options['url']);
 		}
 
+		if (isset($options['base64']) && !$url) {
+			$fullPath = WWW_ROOT . $path;
+			$base64 = base64_encode(file_get_contents($fullPath));
+			$ext = pathinfo($fullPath, PATHINFO_EXTENSION);
+			$type = $this->response->getMimeType($ext);
```

Comment:
This now uses the File object, but I will note that `read()` just returns `file_get_contents` and we can just do the `exists()` check ourselves rather than rely on `open()`. (Though happy to change it if required)